sources/@roots/bud-sass/config/stylelint.cjs

```diff
@@ -3,4 +3,12 @@ module.exports = {
     `@roots/bud-stylelint/config`,
     `stylelint-config-recommended-scss`,
   ],
+  rules: {
+    'custom-property-pattern': [
+      '^([a-z][a-z0-9]*)(-{1,2}[a-z0-9]+)*$',
```

Comment:
What is the source for this regular expression?

It seems like all WordPress custom properties follow a form that allows this to be narrowed down a lot more.

All the documentation and devtools searching I've done has led me to believe that wp custom properties will all begin with either `wp--preset` or `wp--custom`.

Just an initial attempt to maybe improve this a little bit:

```suggestion
      '^wp--preset|wp--custom(-{1,2}[a-z0-9]+)*$',
```

But, I supsect we could also incorporate the list of possible properties as well:

https://github.com/roots/bud/blob/c91d22aa73a49412895152a7f0cc2b6a1732ec9d/sources/%40roots/wordpress-theme-json-webpack-plugin/src/theme.ts#L127-L138

Which would be something like `/wp--(preset|custom)--(blocks|border|color|spacing|etc)(-{1,2}[a-z0-9]+)*$/`

That doesn't seem to be like too much extra and would provide an appreciable difference in UX. But, I'm not accounting for any edge cases should they exist, and my regular expressions are untested (I just composed them in the github code review textarea).

But, worth considering and maybe trying to tackle.

Reply:
> What is the source for this regular expression?

It's [`stylelint-config-standard`](https://github.com/stylelint/stylelint-config-standard/blob/f399e875cb46099bcac8aaed165a88d95736d328/index.js#L49-L54). Standard rule doesn't allow double dash `--` what WP is actually using. So I simply extend this rule to allow it.

Reply:
that makes sense. i'd be cool with approving this PR as is so as to not overcomplicate things once the code change is moved over to the proper file.